File: packages/redis/notification/dlq.ts
// Dead Letter Queue
import type { WebsiteStatus } from "@repo/db/client";
import { client } from "../client";

export type DLQ = {
    messageId : string,
    websiteId : string,
    prevStatus : WebsiteStatus,
    currentStatus : WebsiteStatus,
    occurredAt : string,
    retries : string,
    reason : string
}

export const pushToDlq = async ( 
    stream : string, 
    msg : DLQ
) => {
    await client.xAdd(`${stream}:dlq`,"*",msg);
}

File: apps/api-go/internal/domain/dlq.go
package domain

import "time"

// Redis Stream hash payloads do not have any tags

type DLQMessage struct {
	MessageID     string
	WebsiteID     string
	PrevStatus    WebsiteStatus
	CurrentStatus WebsiteStatus
	OccurredAt    time.Time
	Retries       int
	Reason        string
}
